alchemy/src/vercel/project.ts

```diff
@@ -0,0 +1,418 @@
+   */
+  publicSource?: boolean;
+
+  /**
+   * The name of a directory or relative path to the source code of your project. When `null` is used it will default to the project root
+   */
+  rootDirectory?: string;
+
+  /**
+   * The region to deploy Serverless Functions in this project
+   */
+  serverlessFunctionRegion?: string;
+
+  /**
+   * Specifies whether Zero Config Failover is enabled for this project.
+   */
+  serverlessFunctionZeroConfigFailover?: boolean;
+
+  /**
+   * OpenID Connect JSON Web Token generation configuration.
+   */
+  oidcTokenConfig?: {
+    /**
+     * Whether or not to generate OpenID Connect JSON Web Tokens.
+     */
+    enabled: boolean;
+
+    /**
+     * team: `https://oidc.vercel.com/[team_slug]` global: `https://oidc.vercel.com`
+     */
+    issuerMode: "team" | "global";
+  };
+
+  /**
+   * Opt-in to skip deployments when there are no changes to the root directory and its dependencies
+   */
+  enableAffectedProjectsDeployments?: boolean;
+
+  /**
+   * Resource configuration
+   */
+  resourceConfig?: {
+    /**
+     * Whether fluid is enabled
+     */
+    fluid?: boolean;
+
+    /**
+     * Default regions for functions
+     */
+    functionDefaultRegions?: string[];
+
+    /**
+     * Default timeout for functions
+     */
+    functionDefaultTimeout?: number;
+
+    /**
+     * Default memory type for functions
+     */
+    functionDefaultMemoryType?: "standard_legacy" | "standard" | "performance";
+
+    /**
+     * Whether function zero config failover is enabled
+     */
+    functionZeroConfigFailover?: boolean;
+
+    /**
+     * Whether elastic concurrency is enabled
+     */
+    elasticConcurrencyEnabled?: boolean;
+
+    /**
+     * The build machine type
+     */
+    buildMachineType?: "enhanced" | "ultra";
+  };
+}
+
+/**
+ * Output returned after Project creation/update
+ */
+export interface Project extends Resource<"vercel::Project">, ProjectProps {
+  /**
+   * The ID of the project
+   */
+  id: string;
+
+  /**
+   * The account ID that the project belongs to
+   */
+  accountId: string;
+
+  /**
+   * The time at which the project was created
+   */
+  createdAt: number;
+
+  /**
+   * The time at which the project was last updated
+   */
+  updatedAt: number;
+
+  /**
+   * The latest deployment of the project
+   */
+  latestDeployment?: {
+    /**
+     * The ID of the deployment
+     */
+    id: string;
+
+    /**
+     * The URL of the deployment
+     */
+    url: string;
+  };
+}
+
+export const Project = Resource(
+  "vercel::Project",
+  async function (
+    this: Context<Project>,
+    id: string,
+    { accessToken, ...props }: ProjectProps & { accessToken?: Secret },
+  ): Promise<Project> {
+    switch (this.phase) {
+      case "delete": {
+        const api = await createVercelApi({
+          baseUrl: "https://api.vercel.com/v9",
+          accessToken,
+        });
```

Comment:
can you do this once outside of the `switch`?

Reply:
I'll check again, but each operation (e.g. `PUT`, `PATCH`, `DELETE`, `POST`) has a different supported API version 😓 

I found that hitting `/v10` when that route supports `v9` "still works", but that doesn't mean it'll stay backwards compatible when they eventually up to `/v10`.

If they're all the same, I'll pull this out and DRY it.

Reply:
Confirmed a second time – v9, v10, & v11 APIs are all in use depending on the purpose 😵‍💫